R/data_type.R

```diff
@@ -211,26 +210,29 @@ split_raw_data <- function(df,
   cotrt_types <- setdiff(names(df_list), "single-agent")
   
   control_sa_idx <- which(
-    rowSums(df[, conc_idx, drop = FALSE] == 0) == length(conc_idx)
+    rowSums(df[, ..conc_idx, drop = FALSE] == 0) == length(conc_idx)
   )
   control_sa <- df[control_sa_idx, ]
   untreated_tag <- gDRutils::get_env_identifiers("untreated_tag")
   
   if (length(cotrt_types) > 0) {
     df_list[cotrt_types] <- gDRutils::loop(cotrt_types, function(x) {
-      unique_cotrt <- unique(df_list[[x]][, c(cl, drug_ids[["drug_name"]])])
+      colnames <- c(cl, drug_ids[["drug_name"]])
+      unique_cotrt <- unique(df_list[[x]][, ..colnames])
       unique_cotrt_ctrl <- unique(
         control[
           control[[cl]] %in% unique_cotrt[[cl]] &
             control[[drug_ids[["drug_name"]]]] %in% untreated_tag, 
-        ][, c(cl, drug_ids[["drug_name"]])])
+        ][, ..colnames])
       cotrt_matching <- rbind(unique_cotrt, unique_cotrt_ctrl)
       df_merged <- rbind(df_list[[x]], merge(cotrt_matching, control))
       if (x == "matrix") {
         matrix_data <- rbind(df_merged, df_list[["single-agent"]])
-        matrix_data[, conc_idx][is.na(matrix_data[, conc_idx])] <- 0
-        matrix_data[, codrug_drug_id][is.na(matrix_data[, codrug_drug_id])] <- 
-          untreated_tag[1]
+        for (j in conc_idx)
+          data.table::set(matrix_data, which(is.na(matrix_data[[j]])), j, 0)
+        for (j in codrug_drug_id) {
+          data.table::set(matrix_data, which(is.na(matrix_data[[j]])), j, untreated_tag[1])
```

Comment:
Without looping:
`matrix_data[is.na(get(codrug_drug_id)), (codrug_drug_id) := untreated_tag[1]]`